### ISSUE TITLE
Screen Readers: Create Expenses: The amount field has no name and value announced

### ISSUE THREAD

**Issue by olenamihno** [Expensify/App#77557]
## Action Performed:
Precondition: The User should be logged in

1. Using iOS VoiceOver or Android TalkBack, open the Expensify app
2. Swipe to the Create '+' button and double-tap to activate
3. Swipe to the 'Create Expense' button and double-tap to activate
4. Swipe to the 'Amount' input field and type the amount
5. Note how the field is announced.

## Expected Result:
The field should be announced with its programmatic name and entered value.

## Actual Result:
The amount of field has no name and value announced.

## Area issue was found in:
Create Expenses

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Explicitly set both the accessibilityLabel (the name) and the accessibilityValue (the current content) on the <TextInput> component.
https://reactnative.dev/docs/accessibility#accessibility-properties

## Environment:
Apple iPhone 11 - iOS 17.5.1, H0T Mobile - Native (No mobile browser)

## Screenshots/Videos

https://github.com/user-attachments/assets/f585cf97-80c7-4a31-b4f1-3611057ffa53

<img width="828" height="1792" alt="Image" src="https://github.com/user-attachments/assets/89639373-2c42-41d1-8904-0f486d2d7f86" />

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** No
**Logs:** https://stackoverflow.com/c/expensify/questions/4856  NA
**Notes/Photos/Videos:** Any additional supporting documentation  Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7006095
**Issue reported by:** NA
**Slack conversation:** NA
